src/software/ai/navigator/obstacle/obstacle.h

```diff
@@ -1,171 +1,19 @@
-     * velocity of the robot
-     *
-     * @return a six-sided Polygon to represent the boundary around the obstacle
-     */
-    static Obstacle createRobotObstacleWithScalingParams(const Robot& robot,
-                                                         double radius_cushion_scaling,
-                                                         double velocity_cushion_scaling);
+#include "software/new_geom/segment.h"
 
-    /*
-     * Gets the boundary polygon around the given robot obstacle that other robots
-     * should not enter with a buffer additively expanded by
-     * additional_radius_cushion_buffer and additional_velocity_cushion_buffer
-     *
-     * The radius cushion is a hexagon such that the centre to side distance is (minimum
-     * distance needed for two robots to clear each other + radius buffer)
-     *
-     * The velocity cushion is a rectangular projection whose length is (velocity +
-     * velocity buffer) and whose width is the diameter of the radius cushion
-     *
-     * @param robot robot to create obstacle boundary polygon around
-     * @param additional_radius_cushion_buffer additively increases the radius of the
-     * obstacle by specified amount
-     * @param additional_velocity_cushion_buffer additively increases the cushion of
-     * velocity of the robot
-     * @param enable_velocity_cushion if true, enables the cushion of the velocity
-     * of the robot as component of the obstacle
-     *
-     * @return a six-sided Polygon to represent the boundary around the obstacle
-     */
-    static Obstacle createRobotObstacleWithBufferParams(
-        const Robot& robot, bool enable_velocity_cushion,
-        double additional_radius_cushion_buffer,
-        double additional_velocity_cushion_buffer);
-
-    /**
-     * Circle obstacle around ball with additional_radius_cushion_buffer
-     *
-     * @param ball                              ball to make obstacle around
-     * @param additional_radius_cushion_buffer  extra buffer around obstacle
-     *
-     * @return obstacle around the ball
-     */
-    static Obstacle createCircularBallObstacle(const Ball& ball,
-                                               double additional_radius_cushion_buffer);
-
-
-    const std::optional<Polygon> getBoundaryPolygon() const;
-
-    const std::optional<Circle> getBoundaryCircle() const;
-
-    bool containsPoint(const Point& point) const;
-
-    bool intersects(const Segment& segment) const;
-
-    bool isPolygon() const;
-
-   private:
-    static Obstacle createRobotObstacleFromPositionAndRadiusAndVelocity(
-        Point position, double radius_cushion, Vector velocity_cushion_vector,
-        bool enable_velocity_cushion);
-    static double getRadiusCushionForHexagon(double radius);
-    std::optional<Polygon> _polygon;
-    std::optional<Circle> _circle;
-};
+/**
+ * An obstacle is an area to avoid for navigation
+ */
+using Obstacle = std::shared_ptr<ShapeObstacle>;
```

Comment:
What was the purpose of changing Obstacle to be a pointer to `ShapeObstacle`? I think naming the type `Obstacle` will cause confusion with the pointer semantics. I would prefer a typedef like `ObstaclePtr` and keep the previous `Obstacle` type as is, so that the type name at least tells us this is a pointer.